### DISCUSSION TITLE
List NuGet package dependencies

### DISCUSSION THREAD

**Discussion by christianbumann** [cake-build/cake#3340] [Extension requests]
Hi

It is possible to get a NugetList, but it would be nice if there are also the dependencies listed, or a function exists where the dependencies can be loaded without restoring the packages.
I didn't find such code

Regards Chris

**Comment by nils-a** [comment]
@christianbumann Can you elaborate a bit? Are you searching for a way to list the dependencies of a Cake file?

**Reply by devlead** [reply]
@christianbumann dependencies for a project or a Cake file?

**Reply by christianbumann** [reply]
I mean the dependencies from any NuGet package. For example
[https://www.nuget.org/packages/Cake.Compression/](https://www.nuget.org/packages/Cake.Compression/)
The listed dependencies would be SharpZibLib

[https://cakebuild.net/api/Cake.Common.Tools.NuGet.List/NuGetList/](https://cakebuild.net/api/Cake.Common.Tools.NuGet.List/NuGetList/) only contains a list of [https://cakebuild.net/api/Cake.Common.Tools.NuGet.List/NuGetListItem/](https://cakebuild.net/api/Cake.Common.Tools.NuGet.List/NuGetListItem/), on this item it would be great to be able to receive the dependencies.

It's no longer relevant for me, but it would be nice to have this possibility...

**Comment by pascalberger** [comment]
The NuGet aliases / NuGetTool in Cake.Common is just a light tool wrapper around `nuget.exe`. Implementing additional business logic in it is not something we'll provide in these wrappers. If additional logic is required this can be implemented as part of an addin.

@christianbumann  [GetPackageReferences](https://cakebuild.net/api/Cake.ExtendedNuGet/ExtendedNuGetAliases/378671A7) from the [Cake.ExtendedNuGet](https://cakebuild.net/extensions/cake-extendednuget/) addin sounds like it might do what you're looking for.

**Reply by christianbumann** [reply]
@pascalberger Thank you for the answer. I'll keep it in mind if I need this extension sometimes.
